### PR TITLE
Add Find Command Functionality

### LINKED ISSUE #16: Find notes based on title of notes
As a user I can search for my notes based on its name

### PR DESCRIPTION
**Pull request by R-Ramana**
Closes #16 

- Finds notes based on the title [stores in an ArrayList]
- Prints out ArrayList [from the InterfaceManager]

### CODE REVIEW

File: src/main/java/seedu/duke/util/Parser.java [hunk 1]
```diff
@@ -237,6 +237,14 @@ private Command prepareDeleteNote(String userMessage) {
         }
     }
 
+    private Command prepareFind(String userMessage) throws SystemException {
```
Comment (Chongjx) [inline src/main/java/seedu/duke/util/Parser.java:240]: Is it possible to catch the exception with this function and return a new IncorrectCommand("Missing info" or "No keyword") here

File: src/main/java/seedu/duke/command/FindCommand.java [hunk 2]
```diff
@@ -22,6 +29,11 @@ public FindCommand(String keywords) {
 
     @Override
     public String execute() {
+        ArrayList<Note> filteredNotes = (ArrayList<Note>) notebook.getNotes().stream()
+                .filter((s) -> s.getTitle().contains(keywords))
+                .collect(Collectors.toList());
+
+        printFindNotesMessage(filteredNotes);
         return null;
```
Comment (Chongjx) [inline src/main/java/seedu/duke/command/FindCommand.java:37]: I am thinking that the printFindNotesMessage in InterfaceManager can be a static function and it returns a string of the formatted note. This is to consolidate all the printing to be only done inside the main runLoop function

File: src/main/java/seedu/duke/ui/InterfaceManager.java [hunk 3]
```diff
@@ -61,4 +63,15 @@ private void prints(Notebook notebook) {
...
+    public static void printFindNotesMessage(ArrayList<Note> filteredNotes) {
+        if (filteredNotes.isEmpty()) {
+            System.out.println(" Keyword returns no search result! Try other keywords");
+        } else {
+            System.out.println(" Here are the matching notes in your list:");
+            for (int i = 0; i < filteredNotes.size(); i++) {
+                System.out.println(" " + (i + 1) + "." + filteredNotes.get(i).toString());
+            }
+        }
+    }
```
Comment (Chongjx) [inline src/main/java/seedu/duke/ui/InterfaceManager.java:76]: Make this function return the string instead of printing them out directly

**Review by Chongjx — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/duke/util/Parser.java (hunk 1 above)
Comment (R-Ramana) [reply src/main/java/seedu/duke/util/Parser.java]: Addressed 👍

File: src/main/java/seedu/duke/command/FindCommand.java (hunk 2 above)
Comment (R-Ramana) [reply src/main/java/seedu/duke/command/FindCommand.java]: Will not print, will return a String instead

File: src/main/java/seedu/duke/command/FindCommand.java [hunk 4]
```diff
@@ -29,11 +27,20 @@ public FindCommand(String keywords) {
...
-        return null;
+        for (int i = 0; i < filteredNotes.size(); i++) {
+            notes += (i + 1) + "." + filteredNotes.get(i).toString();
+        }
+
+        if (filteredNotes.isEmpty()) {
+            return "There are no matching notes. Please try another search query.\n";
+        }
+
+        return "Here are the matching notes in your list:\n" + notes;
```
Comment (Chongjx) [inline src/main/java/seedu/duke/command/FindCommand.java:44]: Can use InterfaceManager.LS (line separator) for the \n to standardize all new line

**Review by Chongjx — APPROVED** [review]
Looks good!

**Review by Chongjx — APPROVED** [review]

**Pull request merged**